#!/usr/bin/env python
PACKAGE = "ximea_camera"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("gain", double_t, 0, "Gain in dB", 0, -4, 38)
gen.add("exposure", double_t, 0, "Exposure time in us", 0, 0, 1e+06)
gen.add("aeag", bool_t, 0, "Automatic exposure/gain", False)
gen.add("aeag_roi_offset_x", int_t, 0, "Automatic exposure/gain ROI offset X", 0, 0, 0)
gen.add("aeag_roi_offset_y", int_t, 0, "Automatic exposure/gain ROI offset Y", 0, 0, 0)
gen.add("aeag_roi_width", int_t, 0, "Automatic exposure/gain ROI Width", 2592, 0, 2592)
gen.add("aeag_roi_height", int_t, 0, "Automatic exposure/gain ROI Height", 1944, 0, 1944)
gen.add("bpc", bool_t, 0, "Correction of bad pixels", False)
gen.add("auto_wb", bool_t, 0, "Automatic white balance", False)
gen.add("width", int_t, 0, "Width of the Image provided by the device (in pixels)", 2592, 4, 2592)
gen.add("height", int_t, 0, "Height of the Image provided by the device (in pixels)", 1944, 2, 1944)
gen.add("offsetX", int_t, 0, "Horizontal offset from the origin to the area of interest (in pixels)", 0, 0, 2592)
gen.add("offsetY", int_t, 0, "Vertical offset from the origin to the area of interest (in pixels)", 0, 0, 1944)

exit(gen.generate(PACKAGE, "ximea_camera", "xiAPI"))

